File: local/runner/tests/resources/r/analysis/recipe_stats.r
suppressPackageStartupMessages(library(tidyverse))
library(syre)

# initialize project
db <- database()

# get recipe name
recipe <- root(db)

# get noise-stats data
noise_stats <- db |> find_assets(type = "noise-stats")

# read data
df <- list()
for (stat in noise_stats) {
  tdf <- stat@file |> read_csv()
  df[[length(df) + 1]] <- tdf
}

# combine all the data frames into one
df <- do.call(rbind, df)

# calculate the mean of the 'mean' column
mean_of_mean <- mean(df$mean)

# calculate the mean of the 'std' column
mean_of_std <- mean(df$std)

# create a new data frame with mean_of_mean and mean_of_std
stats_df <- data.frame(
  mean = mean_of_mean,
  std = mean_of_std
)

name <- paste(recipe@name, "Statistics")

# create new asset for the statistics
stats_path <- db |> add_asset(
  "recipe-stats.csv",
  name = name,
  type = "recipe-stats"
)

# save the statistics to the new asset
stats_df |> write.csv(stats_path, row.names = FALSE)

File: local/runner/tests/resources/py/analysis/recipe_stats.py
# include packages
import pandas as pd
import syre

# initialize database
db = syre.Database()

# get recipe container
recipe = db.root

# get noise statistics data
noise_stats = db.find_assets(type='noise-stats')

# create combined dataframe 
df = []
for stat in noise_stats:
	# read data for each batch
	tdf = pd.read_csv( 
		stat.file, 
		names = (stat.metadata['batch'],), 
		index_col = 0, 
		header = 0 
	)
	
	df.append(tdf)

df = pd.concat(df, axis = 1)

# compute recipe statistics
mean = df.loc['mean'].mean() 
std = df.loc['std'].pow(2).sum()/4 

stats = pd.DataFrame([mean, std], index = ('mean', 'std'))

# export recipe statistics
stats_path = db.add_asset(
    'recipe-stats.pkl',
	name='{} Statistics'.format(recipe.name),
	type='recipe-stats'
)

stats.to_pickle(stats_path)
